src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -788,45 +718,54 @@ bool ExecutorWorker::checkLiteSyncInfoForCreate(SyncOpPtr syncOp, const SyncPath
         }
     }
 
-    return true;
+    return ExitCode::Ok;
 }
 
-ExitCode ExecutorWorker::createPlaceholder(const SyncPath &relativeLocalPath, ExitCause &exitCause) {
-    exitCause = ExitCause::Unknown;
-
+ExitInfo ExecutorWorker::createPlaceholder(const SyncPath &relativeLocalPath) {
     SyncFileItem syncItem;
     if (!_syncPal->getSyncFileItem(relativeLocalPath, syncItem)) {
         LOGW_SYNCPAL_WARN(_logger, L"Failed to retrieve SyncFileItem associated to item: "
                                            << Utility::formatSyncPath(relativeLocalPath).c_str());
-        exitCause = ExitCause::InvalidSnapshot;
-        return ExitCode::DataError;
+        return {ExitCode::DataError, ExitCause::InvalidSnapshot};
+    }
+
+    bool exists = false;
+    IoError ioError = IoError::Success;
+    SyncPath absoluteLocalPath = _syncPal->localPath() / relativeLocalPath;
+    if (!IoHelper::checkIfPathExists(absoluteLocalPath, exists, ioError)) {
```

Comment:
This is already checked in VfsWin::createPlaceholder & VfsMac::createPlaceholder.

Reply:
Ideally, I should have modified VfsXX::createPlaceholder to return an ExitInfo.
If we don’t have permission to create the placeholder on Windows, we might still be able to create it (why this happens, I’m not sure). However, we then won’t be able to set the pin state.
To address this, I check if the file exists before attempting to create the placeholder. This way, if processCreateOrConvertToPlaceholderError finds that the file exists, it indicates that setting the pin state failed.